src/components/ScenarioComponents/CardsComponents/DataCard.tsx

```diff
@@ -2,57 +2,52 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import Box from '@mui/material/Box';
-import React, {useEffect, useMemo, useState} from 'react';
+import React, {Dispatch, useEffect, useMemo, useState} from 'react';
 import MainCard from './MainCard/MainCard';
 import FiltersContainer from './GroupFilter/FiltersContainer';
-import {FilterValue} from 'types/card';
+import {FilterValues} from 'types/card';
 import {GroupFilter} from 'types/group';
 import {Localization} from 'types/localization';
-import {Dictionary} from 'util/util';
 
 interface DataCardProps {
   /** A unique identifier for the card.*/
-  index: number;
+  id: string;
 
   /** A dictionary of compartment values associated with the card.*/
-  compartmentValues: Dictionary<number> | null;
+  compartmentValues: Record<string, number | null> | null;
 
   /** A dictionary of start values used for calculating the rate. This determines whether the values have increased, decreased, or remained the same. */
-  startValues: Dictionary<number> | null;
+  referenceValues: Record<string, number> | null;
 
   /** The title of the card.*/
-  label: string;
+  title: string;
 
   /** A boolean indicating whether the compartments are expanded.*/
   compartmentsExpanded: boolean;
 
-  /** An array of compartments.*/
-  compartments: string[];
-
   /** The compartment that is currently selected.*/
-  selectedCompartment: string;
+  selectedCompartmentId: string | null;
 
   /** A boolean indicating whether the scenario is selected.*/
-  selectedScenario: boolean;
+  isSelected: boolean;
 
   /** The color of the card.*/
   color: string;
 
-  /** An array of active scenarios.*/
-  activeScenarios: number[] | null;
+  /** if this scenario is active.*/
+  isActive: boolean;
 
   /** A dictionary of filter values. This is an array of objects, each containing a title and a dictionary of numbers representing
    * the filtered information to be displayed, it's used a disctionary because each card has to have the same amount of filter. */
-  filterValues?: Dictionary<FilterValue[]> | null;
+  filterValues?: Record<string, FilterValues[]> | null;
 
   /** A function to set the selected scenario.*/
-  setSelectedScenario: React.Dispatch<React.SetStateAction<number | null>>;
+  setSelected: Dispatch<{id: string; state: boolean}>;
 
-  /** A function to set the active scenarios.*/
-  setActiveScenarios: React.Dispatch<React.SetStateAction<number[] | null>>;
+  /** A function to set the active scenario.*/
+  setActive: Dispatch<{id: string; state: boolean}>;
 
-  /** The number of the selected scenario.*/
-  numberSelectedScenario: number | null;
+  hide: Dispatch<string>;
```

Comment:
Missing comment.
We should probably rename this to remove (see comment on `CardContainer.tsx`).
```suggestion
  /** A function to hide/remove the data card */
  hide: Dispatch<string>;
```